test/peaks/yearn/YVaultController.test.js

```diff
@@ -0,0 +1,165 @@
+                reverted = true
+                assert.equal(e.reason, 'NOT_OWNER')
+            }
+            assert.equal(reverted, true)
+        })
+    })
+
+    it('Earn', async () => {
+        this.amounts = [100, 100, 100, 100]
+        const tasks = []
+        for (let i = 0; i < n_coins; i++) {
+            this.amounts[i] = toBN(this.amounts[i]).mul(toBN(10 ** this.decimals[i]))
+            tasks.push(this.reserves[i].approve(this.yVaultZap.address, this.amounts[i]))
+        }
+        await Promise.all(tasks)
+        await this.yVaultZap.mint(this.amounts, '0')
+        assert.equal(fromWei(await this.dusd.balanceOf(this.owner)), '400')
+        assert.equal(fromWei(await this.yCrv.balanceOf(this.controller.address)), '0')
+        assert.equal(fromWei(await this.yVault.balanceOf(this.controller.address)), '380')
+        console.log(fromWei(await this.yVault.balanceOf(this.controller.address)))
+    })
+
+    // Peak tests
+    describe('Only Peak', async () => {
+
+        it('withdraw(): peak withdraw amount from controller', async () => {
+            const dusd = toWei('10') 
+            await this.yVaultPeak.redeemInYusd(dusd, '0')
+            const dusd_balance = await this.dusd.balanceOf(this.owner)
+            const yUSD_balance = await this.yVault.balanceOf(this.owner)
+            const controller_balance = await this.yVault.balanceOf(this.controller.address)
+            const peak_balance = await this.yCrv.balanceOf(this.yVaultPeak.address)
+            console.log('DUSD: ' + fromWei(dusd_balance))
+            console.log('yUSD: ' + fromWei(yUSD_balance))
+            console.log('Controller: ' + fromWei(controller_balance))
+            console.log('Peak: ' + fromWei(peak_balance))
+            assert.equal(fromWei(dusd_balance), '390')
+            assert.equal(fromWei(yUSD_balance), '10')
+            assert.equal(fromWei(controller_balance), '370')
+            assert.equal(fromWei(peak_balance), '20')
+        })
+
+        it('withdraw(): peak withdraw entire controller balance', async () => {
+            const dusd = toWei('390')
+            await this.yVaultPeak.redeemInYusd(dusd, '0')
+            const dusd_balance = await this.dusd.balanceOf(this.owner)
+            const yUSD_balance = await this.yVault.balanceOf(this.owner)
+            const controller_balance = await this.yVault.balanceOf(this.controller.address)
+            const peak_balance = await this.yCrv.balanceOf(this.yVaultPeak.address)
+            console.log('DUSD: ' + fromWei(dusd_balance))
+            console.log('yUSD: ' + fromWei(yUSD_balance))
+            console.log('Controller: ' + fromWei(controller_balance))
+            console.log('Peak: ' + fromWei(peak_balance))
+            assert.equal(fromWei(dusd_balance), '0')
+            assert.equal(fromWei(yUSD_balance), '380')
+            assert.equal(fromWei(controller_balance), '0')
+            assert.equal(fromWei(peak_balance), '20')
+        }) 
```

Comment:
Right, so this is the bug I was talking about. The controller might not have the complete amount of yUSD required but might have yCrv lying around. In that case, the contract should make a best effort to convert yCrv to yUSD and give that to user. You wanna roll out the fix in the same PR? :D